packages/@aws-cdk/cfnspec/build-tools/patch-stack.ts

```diff
@@ -0,0 +1,232 @@
+
+/**
+ * Apply a JSON patch stack into the given target file
+ *
+ * The sources can be taken from one or more directories.
+ */
+import * as path from 'path';
+import * as fastJsonPatch from 'fast-json-patch';
+import * as fs from 'fs-extra';
+
+export interface PatchOptions {
+  readonly quiet?: boolean;
+}
+
+/**
+ * Apply a stack to an existing model
+ */
+export async function loadStack(sourceDirectory: string, options: PatchOptions = {}) {
+  const targetObject: any = {};
+
+  const files = await fs.readdir(sourceDirectory);
+  for (const file of files.sort()) {
+    const fullFile = path.join(sourceDirectory, file);
+    const relName = path.relative(process.cwd(), fullFile);
+
+    let data;
+    if ((await fs.stat(fullFile)).isDirectory()) {
+      data = await loadStack(fullFile, options);
+    } else if (file.endsWith('.json')) {
+      data = await fs.readJson(fullFile);
+    } else {
+      continue;
+    }
+
+    if (file.indexOf('patch') === -1) {
+      if (!options.quiet) {
+        // eslint-disable-next-line no-console
+        console.log(relName);
+      }
+
+      // Copy properties from current object into structure, adding/overwriting whatever is found
+      merge(targetObject, data, []);
+    } else {
+      // Apply the loaded file as a patch onto the current structure
+      patch(targetObject, data, (x) => {
+        if (!options.quiet) {
+          // eslint-disable-next-line no-console
+          console.log(`${relName}: ${x}`);
+        }
+      });
+    }
+  }
+  return targetObject;
+}
+
+/**
+ * Combine a number of stacks and write a file
+ *
+ * Stacks are loaded individually, and merged afterwards.
+ */
+export async function resolveStack(targetFile: string, sourceDirectory: string, options: PatchOptions = {}) {
+  const stack = await loadStack(sourceDirectory, options);
+  await writeSorted(targetFile, stack);
+}
+
+export async function writeSorted(targetFile: string, data: any) {
+  // eslint-disable-next-line @typescript-eslint/no-require-imports
+  const sortJson = require('sort-json');
```

Comment:
Any particular reason for this inlined require (both here and in printSorted)? Is there an implementation reason, or is it just stylistic?

Reply:
It doesn't have a definition file so it needs to be the bad import. But yeah, I can lift it.